compiler/rustc_trait_selection/src/traits/select/mod.rs

```diff
@@ -998,7 +1004,7 @@ impl<'cx, 'tcx> SelectionContext<'cx, 'tcx> {
                 ty::PredicateKind::Ambiguous => Ok(EvaluatedToAmbig),
                 ty::PredicateKind::Clause(ty::ClauseKind::ConstArgHasType(ct, ty)) => {
                     match self.infcx.at(&obligation.cause, obligation.param_env).eq(
-                        DefineOpaqueTypes::No,
+                        DefineOpaqueTypes::Yes,
```

Comment:
Needs the same comment as above.

Reply:
ah right, I thought it was only for generic const params, but it's obviously also needed for anon consts in where bounds